gpage.h

```diff
@@ -222,16 +235,19 @@ namespace gcpp {
 	//
 	inline
 	bool gpage::contains(gsl::not_null<const byte*> p) const noexcept {
-		return &storage[0] <= p && p < &storage[total_size - 1];
+		//  Use std::less<> to compare (possibly unrelated) pointers portably
+		auto const cmp = std::less<>{};
+		auto const ext = extent();
+		return !cmp(p, ext.data()) && cmp(p, ext.data() + ext.size());
```

Comment:
Some other wise person noted exactly this problem with the lack of semantic requirements on `less` and friends and submitted [LWG 2562](http://cplusplus.github.io/LWG/lwg-active.html#2562) which requires:

1. `std::less<T*>{}(x, y) == x < y` when `x < y` is defined
2. `std::equal<T*>::operator()`, `std::greater<T*>::operator()`, ... must all induce consistent orders.
3. `std::less<void>{}(x, y)` et al. induce the same ordering as the pointer specializations if the call results in a comparison of pointer values, effectively extending the resolution of [LWG 2450](http://cplusplus.github.io/LWG/lwg-defects.html#2450).